packages/mui-base/src/Menu/Root/menuReducer.ts

```diff
@@ -0,0 +1,45 @@
+import { ListActionTypes, listReducer, moveHighlight } from '../../useList';
+import { MenuReducerAction, MenuActionTypes, MenuReducerState } from './useMenuRoot.types';
+
+export function menuReducer(state: MenuReducerState, action: MenuReducerAction): MenuReducerState {
+  switch (action.type) {
+    case ListActionTypes.itemHover:
+      return state;
+
+    case MenuActionTypes.toggle:
+      return {
+        ...state,
+        open: !state.open,
+      };
+
+    case MenuActionTypes.open: {
+      const updateHighlight = action.event instanceof KeyboardEvent;
+
+      return {
+        ...state,
+        open: true,
+        highlightedValue: updateHighlight
+          ? moveHighlight(null, 'start', state.items, state.settings)
+          : state.highlightedValue,
+      };
```

Comment:
What is the purpose of this reducer? `useListNavigation` handles the highlight (focus) all by itself, so this is likely conflicting and causing the possible bugs

Reply:
Without handling this here I had to press the right arrow twice to actually focus a first element in the menu.